x/stake/client/rest/query.go

```diff
@@ -6,26 +6,25 @@ import (
 	"net/http"
 
 	"github.com/gorilla/mux"
-	"github.com/tendermint/go-crypto/keys"
 
 	"github.com/cosmos/cosmos-sdk/client/context"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/wire"
 	"github.com/cosmos/cosmos-sdk/x/stake"
 )
 
-// RegisterRoutes - Central function to define routes that get registered by the main application
-func RegisterRoutes(ctx context.CoreContext, r *mux.Router, cdc *wire.Codec, kb keys.Keybase) {
-	r.HandleFunc("/stake/{delegator}/bonding_status/{candidate}", BondingStatusHandlerFn("stake", cdc, kb, ctx)).Methods("GET")
+func registerQueryRoutes(ctx context.CoreContext, r *mux.Router, cdc *wire.Codec) {
+	r.HandleFunc("/stake/{delegator}/bonding_status/{candidate}", bondingStatusHandlerFn("stake", cdc, ctx)).Methods("GET")
+	r.HandleFunc("/stake/candidates", candidatesHandlerFn("stake", cdc, ctx)).Methods("GET")
 }
 
-// BondingStatusHandlerFn - http request handler to query delegator bonding status
-func BondingStatusHandlerFn(storeName string, cdc *wire.Codec, kb keys.Keybase, ctx context.CoreContext) http.HandlerFunc {
+// bondingStatusHandlerFn - http request handler to query delegator bonding status
+func bondingStatusHandlerFn(storeName string, cdc *wire.Codec, ctx context.CoreContext) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		// read parameters
 		vars := mux.Vars(r)
 		delegator := vars["delegator"]
-		validator := vars["validator"]
+		candidate := vars["candidate"]
```

Comment:
we only have validators now - no more candidates

Reply:
That sounds like a pretty major change, can you elaborate? If we don't have candidates then I don't understand how we delegate to potential validators who aren't in the top 100.

Reply:
Does this mean users now have to discover candidates out-of-band, and will have to enter an address when delegating? If so, this changes the UI in a huge way and it would have been nice for the Voyager team to be in the loop for this decision.

Reply:
LOL I'm sorry guys - sometimes I live in my head and forget that other people aren't 100% clued in to everything that I've ever thought :) thanks for the clarity bucky! hehehe